system/jhtree/jhtree.cpp

```diff
@@ -1002,7 +1002,7 @@ void CKeyIndex::init(KeyHdr &hdr, bool isTLK, bool allowPreload)
                 cacheNodes(nodeCache, rootPos, isTLK);
             }
         }
-        rootNode = nodeCache->getNode(this, iD, rootPos, NULL, isTLK);
+        rootNode = nodeCache->getNode(this, iD, rootPos, NodeBranch, NULL, isTLK);
```

Comment:
Not sure you can/should assume that the root node is not a leaf. In particular there's a jira open to try to make it into a leaf more often.

Reply:
Yes, because the code you mention is in the following lines...  I'll add a comment.

Reply:
The code below is working around the fact that it has been written out as two layers when it didn't need to be - but there's a JIRA open saying "Why do we even write it as two layers" ?

Reply:
https://track.hpccsystems.com/browse/HPCC-24708

Reply:
In particular if we release this change as-is, the referenced Jira will become very hard to fix (because of back-compatibility issues)

Reply:
I have a better fix.